### ISSUE TITLE
[Email received] Gekko is watching poloniex and has detected a new trend, advice is to go short.

### ISSUE THREAD

**Issue by imkane** [askmike/gekko#801]
However, even though in config.js I have:
```
config.trader = {
  enabled: true,
```
And imports, backtesting, etc is working great. **But no actual trade was made.** Maybe I'm confused with how the trader is supposed to work? I received this email while I was sleeping so obviously wasn't around to take Gekko's advice ;)

**Comment by imkane** [comment]
Ahhh I see below in the logs:

`Wanted to buy BTC but the amount is too small (0.000000000000) at Poloniex`

Now I'm just more confused as to how these strategies work and how the amount could be 0 :(

**Comment by dinankou** [comment]
Hello,
Im running a live gekko on poloniex as well.
Received the same notification :

> Wanted to buy BTC but the amount is too small (0.000000000000) at Poloniex

But i didn't give it much importance as it was what the bot was supposed to do (sell ETH and buy BTC). Except that gekko gave the same advice a couple of days ago and traded successfully. So obviously the ETH balance was 0 when he gave the second advice.

Dont know if it's related, but when i restarted gekko yesterday i had a "NaN" warning on both ETH and BTC portfolios status. Looks like #390 to me...

edit : i restarted gekko again this morning, portfolio status is ok now... Might be a poloniex.com issue fixed during the night.

**Comment by askmike** [comment]
@imkane 

> Now I'm just more confused as to how these strategies work and how the amount could be 0 :(

The docs are not really clear on this, but Gekko internally uses a concept of plugins:

The strategy is a plugin that listens to market data and screams either LONG or SHORT (at specific times when appropriate). The email plugin responds to advice coming out of your strategy just the same as the the portfolio manager does (the piece of code that listens to advice and executes exchange orders).

But all these plugins are not tied together: the email plugin is not aware that there is another plugin actually executing exchange orders. Also: you can have a trader (plugin) and a paper trader (plugin) running at the same time, consuming the same LONG and SHORT signals from a strat (this use case is limited).

**Though note that "plugins" are internal concepts within Gekko, if you use the UI (for which a live trader is in development, see #737) you don't have to worry about this and can use a simpler interface (which under the hood configures Gekko with the correct plugins).**

TLDR: the strategy can advice SHORT (and emails will be sent if you have enabled that plugin), regardless of how much money you have on the exchange, the advice is simply to go short.

@dinankou see above, also with #737 the NaN issue is fixed.
